server/fleet/scripts.go

```diff
@@ -215,14 +215,14 @@ func (hsr HostScriptResult) UserMessage(hostTimeout bool) string {
 	case -1:
 		return RunScriptScriptTimeoutErrMsg
 	case -2:
-		return "Scripts are disabled for this host. To run scripts, deploy a Fleet installer with scripts enabled."
+		return RunScriptDisabledErrMsg
 	default:
 		return ""
 	}
 }
 
 func (hsr HostScriptResult) HostTimeout(waitForResultTime time.Duration) bool {
-	return time.Now().After(hsr.CreatedAt.Add(waitForResultTime))
+	return hsr.ExitCode == nil && time.Now().After(hsr.CreatedAt.Add(waitForResultTime))
```

Comment:
let's say you ran the script with `fleetctl`, then you currently see the right error in the terminal, but the wrong error in the UI, is that right?

Reply:
I think it would depend on when fleetctl tries to fetch the script results, see this [comment](https://github.com/fleetdm/fleet/pull/15967#issuecomment-1881779234) for more info about the underlying issue.

Reply:
yeah, I think that'll be the case because the sync endpoint checks the script results right after it receives a response from the host 👍